### ISSUE TITLE
implement gradient checkpointing

### ISSUE THREAD

**Issue by rom1504** [lucidrains/DALLE2-pytorch#221]
This brought huge memory saving at a little compute cost in other places

I think it could be really helpful here too

**Comment by lucidrains** [comment]
@rom1504 i don't think it'll do that much in this setting, but you can try it out with https://github.com/lucidrains/DALLE2-pytorch/commit/be3bb868bfa428822b892709ecd79c70ad6f73fd#diff-038ecede954c29266888cb88e37cc06f61ba2433f8b5142c7b2b2cefde5ed0edR1724

**Comment by lucidrains** [comment]
ok this should be done for resnet blocks, we can add it for attention blocks at some future date
